server/thryve/src/main/kotlin/com/hatchgrid/thryve/authentication/application/query/GetUserSessionQueryHandler.kt

```
@@ -0,0 +1,26 @@
+class GetUserSessionQueryHandler(
+    private val reactiveJwtDecoder: ReactiveJwtDecoder
+) : QueryHandler<GetUserSessionQuery, UserSession> {
+
+    override suspend fun handle(query: GetUserSessionQuery): UserSession = try {
+        val jwt = reactiveJwtDecoder.decode(query.accessToken)?.awaitSingle()
+            ?: throw InvalidTokenException("Invalid access token - decoder returned null")
+        val userId = UUID.fromString(jwt.subject)
+        val email = jwt.claims["email"] as String
+        val roles = jwt.getClaimAsStringList("roles") ?: emptyList()
+        UserSession(userId, email, roles)
+    } catch (e: Exception) {
```

detekt (1.23.8)

[warning] The caught exception is too generic. Prefer catching specific exceptions to the case that is currently handled.

(detekt.exceptions.TooGenericExceptionCaught)